src/crimson/osd/osd.h

```diff
@@ -232,7 +232,7 @@ class OSD final : public crimson::net::Dispatcher,
     Ref<MOSDPGUpdateLogMissingReply> m);
 
 private:
-  crimson::common::Gated gate;
+  crimson::common::gate_per_shard gate;
```

Comment:
Why should the OSD also use this `gate_per_shard`? Wasn't the issue only in AlienStore usage?

Reply:
The class `OSD` is not sharded, but the messages are dispatched from Connections living in all the available cores, so `OSD::ms_dispatch()` was using the shared gate, which looks to be a related issue.